test/lit/nominal-chain.wast

```diff
@@ -0,0 +1,26 @@
+;; RUN: wasm-opt %s -all --nominal -S -o - | filecheck %s
+;; RUN: wasm-opt %s -all --nominal --roundtrip -S -o - | filecheck %s
+
+;; Check that intermediate types in subtype chains are also included in the
+;; output module, even if there are no other references to those intermediate
+;; types.
+
+(module
+
+  ;; CHECK:      (type $root (struct ))
+  ;; CHECK-NEXT: (type $none_=>_ref?|$root| (func (result (ref null $root))))
+  ;; CHECK-NEXT: (type $leaf (struct (field i32) (field i64) (field f32) (field f64)) (extends $twig))
+  ;; CHECK-NEXT: (type $twig (struct (field i32) (field i64) (field f32)) (extends $branch))
+  ;; CHECK-NEXT: (type $branch (struct (field i32) (field i64)) (extends $trunk))
+  ;; CHECK-NEXT: (type $trunk (struct (field i32)) (extends $root))
```

Comment:
did you add these manually? will the auto-update script not trample them later? (we don't test the script on CI)

Reply:
The checks in this file are not generated by the auto-update script, so it shouldn't be an issue.